### ISSUE TITLE
1.5.0 iconv_t problems

### ISSUE THREAD

**Issue by neheb** [gerbera/gerbera#842]
Looks like some of my cast changes broke this :)

```
/home/mangix/devstuff/openwrt/build_dir/target-arc_arc700_uClibc/gerbera-1.5.0/src/util/string_converter.cc: In constructor 'StringConverter::StringConverter(const string&, const string&)':
/home/mangix/devstuff/openwrt/build_dir/target-arc_arc700_uClibc/gerbera-1.5.0/src/util/string_converter.cc:44:43: error: invalid cast from type 'int' to type 'iconv_t' {aka 'long int'}
   44 |     if (cd == reinterpret_cast<iconv_t>(-1)) {
      |                                           ^
/home/mangix/devstuff/openwrt/build_dir/target-arc_arc700_uClibc/gerbera-1.5.0/src/util/string_converter.cc:45:42: error: invalid static_cast from type 'std::nullptr_t' to type 'iconv_t' {aka 'long int'}
   45 |         cd = static_cast<iconv_t>(nullptr);
      |                                          ^
/home/mangix/devstuff/openwrt/build_dir/target-arc_arc700_uClibc/gerbera-1.5.0/src/util/string_converter.cc: In destructor 'virtual StringConverter::~StringConverter()':
/home/mangix/devstuff/openwrt/build_dir/target-arc_arc700_uClibc/gerbera-1.5.0/src/util/string_converter.cc:52:43: error: invalid static_cast from type 'std::nullptr_t' to type 'iconv_t' {aka 'long int'}
   52 |     if (cd != static_cast<iconv_t>(nullptr))

```

**Comment by whyman** [comment]
Are you going to send a fix or?

**Comment by neheb** [comment]
https://github.com/gerbera/gerbera/pull/847

**Comment by whyman** [comment]
Closed by #847
